src/mocks/controllers/productController.js

```diff
@@ -0,0 +1,186 @@
+/* eslint-disable no-plusplus */
+/* eslint-disable no-underscore-dangle */
+import * as mock from 'mocks/mockForTesting';
+import { controllerWrapper } from 'mocks/utils/utilFunction';
+import * as Product from 'mocks/models/productModel';
+import { verifyToken } from 'mocks/models/authenModel';
+
+const getRequestParams = (req) => {
+  const sortBy = req.url.searchParams.getAll('sortBy')[0];
+  const search = req.url.searchParams.getAll('search')[0] || null;
+  return { sortBy, search };
+};
+
+const getTokenFromRequest = (req) =>
+  req.headers._headers.authorization.split(' ')[1];
+
+const getSortFormula = (sortBy) => {
+  switch (sortBy) {
+    case 'PRICE_DECREASE':
+      return (a, b) => b.price - a.price;
+    case 'PRICE_INCREASE':
+      return (a, b) => a.price - b.price;
+    default:
+      // for most recently added item first
+      return (a, b) => b.createdTimestamp - a.createdTimestamp;
+  }
+};
+
+const hasSearch = (title, search) => {
+  const titleWords = title.split(' ');
+  const searchWords = search.split(' ');
+
+  for (let i = 0; i < titleWords.length; i++) {
+    for (let j = 0; j < searchWords.length; j++) {
+      if (titleWords[i].toLowerCase() === searchWords[j].toLowerCase()) {
+        return true;
+      }
+    }
+  }
+  return false;
+};
```

Comment:
I think we don't have to match the exact word. For example, if users want to search `Shoes`, when they type `s`, the `Shoes` should be displayed.
search. Something like `title.toLowerCase().includes(search.toLowerase())` seems fine to me.